### PR TITLE
Add smoke tests for Interpreter Dropdown

### PR DESCRIPTION
**Pull request by sharon-wang**
## Description

- Addresses https://github.com/posit-dev/positron/issues/4124
- Adds new test cases
    - Python interpreter starts and shows running
    - Python interpreter restarts and shows running
    - R interpreter starts and shows running
    - R interpreter stops and shows inactive

### QA Notes

- please add TestRail IDs 😄
- please mark tests for #pr CI runs as appropriate
- `StartInterpreter` was refactored significantly (and renamed to `PositronInterpreterDropdown`), but was only previously used by `new-project.test.ts`

### CODE REVIEW

**Comment by sharon-wang** [comment]
Full test suite passed 🎉, although the push to slack failed:

```
Traceback (most recent call last):
  File "/source/app/script.py", line 109, in <module>
    main()
  File "/source/app/script.py", line 91, in main
    slack_utils.send_slack_msg(
  File "/source/app/utils/slack_utils.py", line 19, in send_slack_msg
    client.chat_postMessage(
  File "/usr/local/lib/python3.10/site-packages/slack/web/client.py", line 1077, in chat_postMessage
    return self.api_call("chat.postMessage", json=kwargs)
  File "/usr/local/lib/python3.10/site-packages/slack/web/base_client.py", line [15](https://github.com/posit-dev/positron/actions/runs/10290770371/job/28481507074#step:24:16)0, in api_call
    return self._sync_send(api_url=api_url, req_args=req_args)
  File "/usr/local/lib/python3.10/site-packages/slack/web/base_client.py", line 241, in _sync_send
    return self._urllib_api_call(
  File "/usr/local/lib/python3.10/site-packages/slack/web/base_client.py", line 378, in _urllib_api_call
    ).validate()
  File "/usr/local/lib/python3.10/site-packages/slack/web/slack_response.py", line [19](https://github.com/posit-dev/positron/actions/runs/10290770371/job/28481507074#step:24:20)4, in validate
    raise e.SlackApiError(message=msg, response=self)
slack.errors.SlackApiError: The request to the Slack API failed.
The server responded with: {'ok': False, 'error': 'not_in_channel'}
```

**Comment by testlabauto** [comment]
> Full test suite passed 🎉, although the push to slack failed:
> 
> ```
> Traceback (most recent call last):
>   File "/source/app/script.py", line 109, in <module>
>     main()
>   File "/source/app/script.py", line 91, in main
>     slack_utils.send_slack_msg(
>   File "/source/app/utils/slack_utils.py", line 19, in send_slack_msg
>     client.chat_postMessage(
>   File "/usr/local/lib/python3.10/site-packages/slack/web/client.py", line 1077, in chat_postMessage
>     return self.api_call("chat.postMessage", json=kwargs)
>   File "/usr/local/lib/python3.10/site-packages/slack/web/base_client.py", line [15](https://github.com/posit-dev/positron/actions/runs/10290770371/job/28481507074#step:24:16)0, in api_call
>     return self._sync_send(api_url=api_url, req_args=req_args)
>   File "/usr/local/lib/python3.10/site-packages/slack/web/base_client.py", line 241, in _sync_send
>     return self._urllib_api_call(
>   File "/usr/local/lib/python3.10/site-packages/slack/web/base_client.py", line 378, in _urllib_api_call
>     ).validate()
>   File "/usr/local/lib/python3.10/site-packages/slack/web/slack_response.py", line [19](https://github.com/posit-dev/positron/actions/runs/10290770371/job/28481507074#step:24:20)4, in validate
>     raise e.SlackApiError(message=msg, response=self)
> slack.errors.SlackApiError: The request to the Slack API failed.
> The server responded with: {'ok': False, 'error': 'not_in_channel'}
> ```

Oops.  I forgot to invite the app to the new channel.  Done now!

**Review by testlabauto — DISMISSED** [review]
Nicely done!  Appreciate the excellent refactor of positronStartInterpreter.ts!

File: test/smoke/src/fixtures/positronConsoleFixtures.ts
Comment (sharon-wang) [inline test/smoke/src/fixtures/positronConsoleFixtures.ts:1]: This file has been replaced by `test/automation/src/positron/fixtures/positronUserSettingsFixtures.ts`

File: test/smoke/src/areas/positron/top-action-bar/interpreter-dropdown.test.ts [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+			it('Disable automatic startup of interpreters', async function () {
+				// Disable automatic startup of interpreters to prevent the auto-started interpreter from
+				// bulldozing the interpreter dropdown selection
+				await userSettings.setUserSetting([
+					'positron.interpreters.automaticStartup',
+					'false',
+				]);
+			});
+		});
```
Comment (sharon-wang) [inline test/smoke/src/areas/positron/top-action-bar/interpreter-dropdown.test.ts:47]: Disables the auto interpreter setup and the app will be restarted between this describe block and the actual test describe block

File: test/smoke/src/areas/positron/top-action-bar/interpreter-dropdown.test.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+				app = this.app as Application;
+				userSettings = new PositronUserSettingsFixtures(app);
+			});
+
+			it('Unset user settings', async function () {
+				// Unset the user setting to re-enable automatic startup of interpreters
+				await userSettings.unsetUserSettings();
+			});
+		});
+	});
```
Comment (sharon-wang) [inline test/smoke/src/areas/positron/top-action-bar/interpreter-dropdown.test.ts:186]: Clears the user settings after the tests have run, to restore the original behaviour

**Review by sharon-wang — COMMENTED** [review]
I've add "setup" and "cleanup" describe blocks with single "tests" to set and unset the auto interpreter startup user setting. I wasn't able to use the `before` and `after` blocks in the existing `describe`, as the app restart timing interferes with the tests (more info in the comments).

Full test suite in progress: https://github.com/posit-dev/positron/actions/runs/10309907612

Hope to see this pass on Windows too! 🤞

File: test/smoke/src/areas/positron/top-action-bar/interpreter-dropdown.test.ts (hunk 1 above)
Comment (sharon-wang) [reply test/smoke/src/areas/positron/top-action-bar/interpreter-dropdown.test.ts]: this has been removed

File: test/smoke/src/areas/positron/top-action-bar/interpreter-dropdown.test.ts (hunk 2 above)
Comment (sharon-wang) [reply test/smoke/src/areas/positron/top-action-bar/interpreter-dropdown.test.ts]: this has been removed

File: test/smoke/src/areas/positron/top-action-bar/interpreter-dropdown.test.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+			app = this.app as Application;
+			interpreterDropdown = app.workbench.positronInterpreterDropdown;
+			positronConsole = app.workbench.positronConsole;
+			desiredPython = process.env.POSITRON_PY_VER_SEL!;
+			desiredR = process.env.POSITRON_R_VER_SEL!;
+		});
+
+		beforeEach(async function () {
+			await positronConsole.waitForReadyOrNoInterpreter();
+		});
```
[the remark this answers is not in the document]
Comment (sharon-wang) [reply test/smoke/src/areas/positron/top-action-bar/interpreter-dropdown.test.ts]: this is removed now

**Review by jonvanausdeln — APPROVED** [review]
LGTM! New timeout works on my machine!  Hopefully the CI pass works!

**Comment by sharon-wang** [comment]
CI is green! ✅ 🎉

**Pull request merged**